### PR TITLE
Add non-Euclidean Poisson fluxes

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

These can be used to solve the Poisson equation on a curved
background metric. We'll need to do that for XCTS solves where
we don't assume conformal flatness.

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/Elliptic/Systems/Poisson/Equations.hpp [hunk 1]
```diff
@@ -59,6 +59,17 @@ void euclidean_fluxes(
...
     const tnsr::i<DataVector, Dim, Frame::Inertial>& field_gradient) noexcept;
 
+/*!
+ * \brief Compute the fluxes \f$F^i=\sqrt{\gamma}\gamma^{ij}\partial_j u(x)\f$
+ * for the curved-space Poisson equation on a metric \f$\gamma_{ij}\f$.
+ */
+template <size_t Dim>
+void noneuclidean_fluxes(
+    gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*> flux_for_field,
+    const tnsr::II<DataVector, Dim, Frame::Inertial>& inv_metric,
```
Comment (nilsdeppe) [inline src/Elliptic/Systems/Poisson/Equations.hpp]: Is this the metric or spatial metric? Your code says metric but your dox say spatial metric.

File: src/Elliptic/Systems/Poisson/Equations.hpp [hunk 2]
```diff
@@ -59,6 +59,17 @@ void euclidean_fluxes(
     gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*> flux_for_field,
     const tnsr::i<DataVector, Dim, Frame::Inertial>& field_gradient) noexcept;
 
+/*!
+ * \brief Compute the fluxes \f$F^i=\sqrt{\gamma}\gamma^{ij}\partial_j u(x)\f$
+ * for the curved-space Poisson equation on a metric \f$\gamma_{ij}\f$.
```
Comment (nilsdeppe) [inline src/Elliptic/Systems/Poisson/Equations.hpp]: is gamma the metric or spatial metric?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
review with @wthrowe and @kidder

File: src/Elliptic/Systems/Poisson/Equations.hpp (hunk 1 above)
Comment (nilsvu) [reply src/Elliptic/Systems/Poisson/Equations.hpp]: pushed a fixup, please take a look

**Comment by nilsvu** [comment]
@nilsdeppe This only needed some docs changes, please check the fixup

**Review by nilsdeppe — COMMENTED** [review]
LGTM. Please squash.

**Comment by kidder** [comment]
@nilsdeppe this has been squashed

**Review by nilsdeppe — APPROVED** [review]

**Comment by nilsdeppe** [comment]
Travis failure is #1730

**Review by kidder — APPROVED** [review]

**Pull request merged**